tools/traffic/workers/blob_reader.go

```diff
@@ -0,0 +1,223 @@
+package workers
+
+import (
+	"context"
+	"crypto/md5"
+	"fmt"
+	"github.com/Layr-Labs/eigenda/api/clients"
+	contractEigenDAServiceManager "github.com/Layr-Labs/eigenda/contracts/bindings/EigenDAServiceManager"
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/Layr-Labs/eigenda/encoding"
+	"github.com/Layr-Labs/eigenda/retriever/eth"
+	"github.com/Layr-Labs/eigenda/tools/traffic/config"
+	"github.com/Layr-Labs/eigenda/tools/traffic/metrics"
+	"github.com/Layr-Labs/eigenda/tools/traffic/table"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	gcommon "github.com/ethereum/go-ethereum/common"
+	"math/big"
+	"sync"
+	"time"
+)
+
+// BlobReader reads blobs from a disperser at a configured rate.
+type BlobReader struct {
+	// The context for the generator. All work should cease when this context is cancelled.
+	ctx *context.Context
+
+	// Tracks the number of active goroutines within the generator.
+	waitGroup *sync.WaitGroup
+
+	// All logs should be written using this logger.
+	logger logging.Logger
+
+	// config contains the configuration for the generator.
+	config *config.WorkerConfig
+
+	retriever   clients.RetrievalClient
+	chainClient eth.ChainClient
+
+	// blobsToRead blobs we are required to read a certain number of times.
+	blobsToRead *table.BlobStore
+
+	generatorMetrics           metrics.Metrics
+	fetchBatchHeaderMetric     metrics.LatencyMetric
```

Comment:
could be cleaner to consolidate these metrics into a struct

Reply:
done